packages/tailwindcss/src/utilities.ts

```diff
@@ -4569,3 +4584,360 @@ export function createUtilities(theme: Theme) {
 
   return utilities
 }
+
+export function createCssUtility(node: AtRule) {
+  let name = node.params
+
+  // Functional utilities. E.g.: `tab-size-*`
+  if (IS_VALID_FUNCTIONAL_UTILITY_NAME.test(name)) {
+    // API:
+    //
+    // - `--value(number)`            resolves a bare value of type number
+    // - `--value([number])`          resolves an arbitrary value of type number
+    // - `--value(--color)`           resolves a theme value in the `color` namespace
+    // - `--value(number, [number])`  resolves a bare value of type number or an
+    //                                arbitrary value of type number in order.
+    //
+    // Rules:
+    //
+    // - If `--value(…)` does not resolve to a valid value, the declaration is
+    //   removed.
+    // - If a `--value(ratio)` resolves, the `--modifier(…)` cannot be used.
+    // - If a candidate looks like `foo-2/3`, then the `--value(ratio)` should
+    //   be used OR the `--value(…)` and `--modifier(…)` must be used. But not
+    //   both.
```

Comment:
is there any valid use case for resolving a ratio like `foo-2/3` as `--value(…)` and `--modifier(…)`? If not I wonder if this might simplify things a bit since we can just determine in advance wether the input looks like a `ratio` and never resolve the `--value(number)` in this case.

Reply:
Sadly there is, if you define your font-sizes as numbers instead of tshirt sizes. E.g.: `text-4/7` could be `text-4/7` (font-size of `4/7` whatever that means) or `text-4` with line-height of `7`.